src/token.rs

```diff
@@ -10,7 +11,10 @@ pub enum Token {
 impl fmt::Display for Token {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Token::Plain(s) | Token::DoubleQuoted(s) | Token::SingleQuoted(s) => write!(f, "{}", s),
+            Token::Meta(s) => write!(f, "{}", s),
```

Comment:
Why not just have a token for each "Meta", without an associated string

Reply:
Hmm, I thought maybe this way would reduce the number of if/else.